zsh-history-substring-search.zsh

```diff
@@ -43,12 +43,44 @@
 # declare global configuration variables
 #-----------------------------------------------------------------------------
 
-: ${HISTORY_SUBSTRING_SEARCH_HIGHLIGHT_FOUND='bg=magenta,fg=white,bold'}
-: ${HISTORY_SUBSTRING_SEARCH_HIGHLIGHT_NOT_FOUND='bg=red,fg=white,bold'}
-: ${HISTORY_SUBSTRING_SEARCH_GLOBBING_FLAGS='i'}
-: ${HISTORY_SUBSTRING_SEARCH_ENSURE_UNIQUE=''}
-: ${HISTORY_SUBSTRING_SEARCH_FUZZY=''}
-: ${HISTORY_SUBSTRING_SEARCH_PREFIXED=''}
+if (( ! $+HISTORY_SUBSTRING_SEARCH_HIGHLIGHT_FOUND )); then
+  typeset -g HISTORY_SUBSTRING_SEARCH_HIGHLIGHT_FOUND='bg=magenta,fg=white,bold'
+fi
+if (( ! $+HISTORY_SUBSTRING_SEARCH_HIGHLIGHT_NOT_FOUND )); then
+  typeset -g HISTORY_SUBSTRING_SEARCH_HIGHLIGHT_NOT_FOUND='bg=red,fg=white,bold'
+fi
+if (( ! $+HISTORY_SUBSTRING_SEARCH_GLOBBING_FLAGS )); then
+  typeset -g HISTORY_SUBSTRING_SEARCH_GLOBBING_FLAGS='i'
+fi
+if (( ! $+HISTORY_SUBSTRING_SEARCH_ENSURE_UNIQUE )); then
+  typeset -g HISTORY_SUBSTRING_SEARCH_ENSURE_UNIQUE=''
+fi
+if (( ! $+HISTORY_SUBSTRING_SEARCH_FUZZY )); then
+  typeset -g HISTORY_SUBSTRING_SEARCH_FUZZY=''
+fi
+if (( ! $+HISTORY_SUBSTRING_SEARCH_PREFIXED )); then
+  typeset -g HISTORY_SUBSTRING_SEARCH_PREFIXED=''
+fi
```

Comment:
I'm also fixing the issue with WARN_CREATE_GLOBAL, as I commented here: https://github.com/zsh-users/zsh-history-substring-search/pull/108#issuecomment-891840776

Reply:
PR #119 seems simpler than this implementation.  I'm leaning toward merging it instead.

Reply:
True, that implementation is simpler. Let me update the PR and use that here.

Reply:
The `typeset -g VAR="${VAR:=value}"` construct does not seem correct to me, as is does the assignment twice: at `${VAR:=value}` and also at `typeset -g VAR="value"`, so I changed it to what seems more correct.